router/io-spec-non-proven-lemmas.gobra

```diff
@@ -74,39 +74,33 @@ func absPktFutureLemma(dp io.DataPlaneSpec, raw []byte) {
 	assert len(pkt.CurrSeg.Future) > 0
 }
 
-/*
 ghost
-preserves dp.Valid()
-preserves acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), R55)
-preserves validPktMetaHdr(raw)
-ensures let LeftSeg := absPkt(dp, raw).LeftSeg in
-	LeftSeg != none[io.IO_seg2] ==>
-		len(get(LeftSeg).History) == 0 && len(get(LeftSeg).Future) > 0
+requires  len(oldPkt.CurrSeg.Future) > 0
+requires  AbsValidateIngressIDConstraint(oldPkt, ingressID)
+requires  newPkt == AbsUpdateNonConsDirIngressSegID(oldPkt, ingressID)
+ensures   AbsValidateIngressIDConstraint(newPkt, ingressID)
 decreases
-func absPktNextSegmentHistory(dp io.DataPlaneSpec, raw []byte) {
-	pkt := reveal absPkt(dp, raw)
-	reveal validPktMetaHdr(raw)
-	headerOffset := slayers.GetAddressOffset(raw)
-	assert forall k int :: {&raw[headerOffset:headerOffset+scion.MetaLen][k]} 0 <= k && k < scion.MetaLen ==> &raw[headerOffset:headerOffset+scion.MetaLen][k] == &raw[headerOffset + k]
-	hdr := (unfolding acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), R56) in
-			binary.BigEndian.Uint32(raw[headerOffset:headerOffset+scion.MetaLen]))
-	metaHdr := scion.DecodedFrom(hdr)
-	currINFIdx := int(metaHdr.CurrINF)
-	currHFIdx := int(metaHdr.CurrHF)
-	seg1Len := int(metaHdr.SegLen[0])
-	seg2Len := int(metaHdr.SegLen[1])
-	seg3Len := int(metaHdr.SegLen[2])
-	segLen := scion.LengthOfCurrSeg(currHFIdx, seg1Len, seg2Len, seg3Len)
-	prevSegLen := scion.LengthOfPrevSeg(currHFIdx, seg1Len, seg2Len, seg3Len)
-	numINF := scion.NumInfoFields(seg1Len, seg2Len, seg3Len)
-	offset := scion.HopFieldOffset(numINF, 0, headerOffset)
-	assert pkt.LeftSeg == reveal scion.LeftSeg(raw, currINFIdx + 1, seg1Len, seg2Len , seg3Len, headerOffset)
-	if(pkt.LeftSeg != none[io.IO_seg2]) {
-		assert len(get(pkt.LeftSeg).Future) > 0
-		assert len(get(pkt.LeftSeg).History) == 0
-	}
+func AbsValidateIngressIDLemma(oldPkt io.IO_pkt2, newPkt io.IO_pkt2, ingressID option[io.IO_ifs]) {
+	reveal AbsValidateIngressIDConstraint(oldPkt, ingressID)
+	reveal AbsUpdateNonConsDirIngressSegID(oldPkt, ingressID)
+	reveal AbsValidateIngressIDConstraint(newPkt, ingressID)
+}
+
+ghost
+requires  len(oldPkt.CurrSeg.Future) == 1
+requires  oldPkt.LeftSeg != none[io.IO_seg2]
+requires  len(get(oldPkt.LeftSeg).Future) > 0
+requires  len(get(oldPkt.LeftSeg).History) == 0
+requires  AbsValidateIngressIDConstraint(oldPkt, ingressID)
+requires  newPkt == AbsDoXover(oldPkt)
+ensures   AbsValidateIngressIDConstraintXover(newPkt, ingressID)
+decreases
+func AbsValidateIngressIDXoverLemma(oldPkt io.IO_pkt2, newPkt io.IO_pkt2, ingressID option[io.IO_ifs]) {
+	reveal AbsValidateIngressIDConstraint(oldPkt, ingressID)
+	reveal AbsDoXover(oldPkt)
+	reveal AbsValidateIngressIDConstraintXover(newPkt, ingressID)
```

Comment:
the lemmas that are currently proven should be moved to a different file (we can create one called `bytes-pkts-lemmas.gobra`)